### ISSUE TITLE
BootBuildImage's archiveFile property is annotated as an @Input but should be annotated as an @InputFile

### ISSUE THREAD

**Issue by juergenzimmermann** [spring-projects/spring-boot#32495]
In a project with Spring Boot _3.0.0-M5_ I tried to upgrade from Gradle _7.5.1_ to _8.0.0-milestone-1_, but got this error message:

```
* What went wrong:
A problem was found with the configuration of task ':bootBuildImage' (type 'BootBuildImage').
  - In plugin 'org.springframework.boot' type 'org.springframework.boot.gradle.tasks.bundling.BootBuildImage' property 'archiveFile' has @Input annotation used on property of type 'RegularFileProperty'.

    Reason: A property of type 'RegularFileProperty' annotated with @Input cannot determine how to interpret the file.

    Possible solutions:
      1. Annotate with @InputFile for regular files.
      2. Annotate with @InputFiles for collections of files.
      3. If you want to track the path, return File.absolutePath as a String and keep @Input.

    Please refer to https://docs.gradle.org/8.0-milestone-1/userguide/validation_problems.html#incorrect_use_of_input_annotation for more details about this problem.
```

**Comment by snicoll** [comment]
Thanks for the report but we are aware that our plugin might not work with a milestone of the next Gradle's generation. We are monitoring this and working with the Gradle team so I am going to close this now. We'll work on this in due course.

**Comment by wilkinsona** [comment]
@juergenzimmermann Thanks for trying out Boot's plugin with Gradle 8.0.0-milestone-1. We don't yet support Gradle 8.0 and haven't started testing against it so some incompatibilities are to be expected. It looks as if Gradle 8.0 won't be available until after Spring Boot 3.0.0-RC1 so, depending on the scope of changes that are required to support 8.0, we may not be able to do so until Boot 3.1. In the meantime, we can fix the specific problem that you've reported as the annotation on the `archiveFile` property is incorrect with earlier versions of Gradle too.

**Comment by juergenzimmermann** [comment]
@wilkinsona Thank you for the commenting the circumstances. When you have a fix regarding `@InputFile`. I can test it with my projects. BTW, compiling Kotlin code and also using tools like ktlint, detekt, Dokka are working fine with the 1st milestone of Gradle 8.0.
